File: app/src/main/java/com/alguojian/logcrash/MainActivity.java
package com.alguojian.logcrash;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    List<String> aa;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.button2).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                for (int i = 0; i < 100; i++) {
                    aa.get(i);
                }
                String aa = "123";
                String bb = aa.substring(9, 123);
            }
        });
    }
}

File: app/src/main/java/com/alguojian/app/MainActivity.kt
package com.alguojian.app

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import com.alguojian.logcrash.LogCrashOpenUtils
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    var aa: List<String>? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        button2.setOnClickListener { v: View? ->
            for (i in 0..99) {
                aa!![1111]
            }
            val aa = "123"
            val bb = aa.substring(9, 123)
        }

        LogCrashOpenUtils.open(button,this@MainActivity)
    }
}